cli/command/manifest/create_list.go

```diff
@@ -34,21 +33,18 @@ func newCreateListCommand(dockerCli command.Cli) *cobra.Command {
 }
 
 func createManifestList(dockerCli command.Cli, args []string, opts annotateOpts) error {
-	// Just do some basic verification here, and leave the rest for when the user pushes the list
 	newRef := args[0]
-	targetRef, err := reference.ParseNormalizedNamed(newRef)
+	targetRef, err := normalizeReference(newRef)
 	if err != nil {
 		return errors.Wrapf(err, "error parsing name for manifest list (%s): %v", newRef)
 	}
+
+	// TODO: why is this here?
 	_, err = registry.ParseRepositoryInfo(targetRef)
 	if err != nil {
 		return errors.Wrapf(err, "error parsing repository name for manifest list (%s): %v", newRef)
 	}
 
-	// Check locally for this list transaction before proceeding
-	if _, isDigested := targetRef.(reference.Canonical); !isDigested {
-		targetRef = reference.TagNameOnly(targetRef)
-	}
 	manifestStore := newDefaultStore()
 	list, err := manifestStore.GetList(targetRef.String())
```

Comment:
No, that's not really the responsibility of the `store`. The store can store anything. It's up to each command to validate/normalize the input. `normalizeReference()` does that, and I believe it's called for all user input in these commands.